sdk/storage/Azure.Storage.Files.DataLake/src/RolePermissionsExtensions.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using System.Text;
+using Azure.Storage.Files.DataLake.Models;
+
+namespace Azure.Storage.Files.DataLake
+{
+    /// <summary>
+    /// Extension methods for RolePermissions.
+    /// </summary>
+    public static class RolePermissionsExtensions
+    {
+        /// <summary>
+        /// Parses octal char to RolePermissions.
+        /// </summary>
+        public static RolePermissions ParseOctal(char octalRolePermission)
+        {
+            RolePermissions rolePermissions = RolePermissions.None;
+
+            int value = (int)char.GetNumericValue(octalRolePermission);
+
+            if (value < 0 || value > 7)
+            {
+                throw Errors.MustBeBetweenInclusive(nameof(octalRolePermission), 0, 7);
+            }
+
+            if ((value & 4) > 0)
+            {
+                rolePermissions |= RolePermissions.Read;
+            }
+
+            if ((value & 2) > 0)
+            {
+                rolePermissions |= RolePermissions.Write;
+            }
+
+            if ((value & 1) > 0)
+            {
+                rolePermissions |= RolePermissions.Execute;
+            }
+
+            return rolePermissions;
+        }
+
+        /// <summary>
+        /// Parses symbolic permissions string to RolePermissions.
+        /// </summary>
+        /// <param name="symbolicRolePermissions">String to parse.</param>
+        /// <param name="allowStickyBit">If sticky bit is allowed.</param>
+        /// <returns><see cref="RolePermissions"/>.</returns>
+        public static RolePermissions ParseSymbolic(string symbolicRolePermissions, bool allowStickyBit)
+        {
+            RolePermissions rolePermissions = RolePermissions.None;
+            ArgumentException argumentException = new ArgumentException("Role permission contains an invalid character");
+
+            if (symbolicRolePermissions == null)
+            {
+                throw Errors.ArgumentNull(nameof(symbolicRolePermissions));
+            }
+
+            if (symbolicRolePermissions.Length != 3)
+            {
+                throw new ArgumentException("Role permission must be 3 characters");
+            }
+
+            if (symbolicRolePermissions[0] == 'r')
```

Comment:
Do you need `ToLower`?

Reply:
As far as I know, the symbolic permissions should be lower case.